File: leetcode/common/ListNode.java
package common;

public class ListNode {
    public int val;
    public ListNode next;

    public ListNode() {
    }

    public ListNode(int val) {
        this.val = val;
    }

    public ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }

    /**
     * 用数组初始化一个单链表
     */
    public static ListNode init(int[] arr) {
        if (arr.length == 0) {
            return null;
        }

        ListNode head = new ListNode(arr[0]);
        ListNode current = head;

        for (int i = 1; i < arr.length; i++) {
            current.next = new ListNode(arr[i]);
            current = current.next;
        }

        return head;
    }

    /**
     * 获取值
     */
    public int getVal() {
        return val;
    }

    /**
     * 获取下一个节点
     */
    public ListNode getNext() {
        return next;
    }

    @Override
    public String toString() {
        return "ListNode [val=" + val + ", " + (next != null ? "next=" + next : "") + "]";
    }
}

File: leetcode/common/ListNode.dart
class ListNode {
  int val;
  ListNode? next;

  // Dart 的构造函数
  ListNode([this.val = 0, this.next]);

  // 用数组初始化一个单链表
  static ListNode? init(List<int> arr) {
    if (arr.isEmpty) {
      return null;
    }

    ListNode head = ListNode(arr[0]);
    ListNode current = head;

    for (int i = 1; i < arr.length; i++) {
      current.next = ListNode(arr[i]);
      current = current.next!;
    }

    return head;
  }

  // 获取值
  int get value => val;

  // 获取下一个节点
  ListNode? get nextNode => next;

  @override
  String toString() {
    if (next != null) {
      return 'ListNode(val: $val, next: $next)';
    } else {
      return 'ListNode(val: $val)';
    }
  }
}
